### DISCUSSION TITLE
announcement bar - relative link?

### DISCUSSION THREAD

**Discussion by VladimirFokow** [squidfunk/mkdocs-material#6881] [Q&A] [answered]
Is it possible to use a link to a page in the same docs
in the [announcement bar](https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-header/#announcement-bar), that would work correctly when clicked from any page?

<br />

For example, if I have this in my `main.html`:
```html
{% extends "base.html" %}

{% block announce %}
  <a href="path/to/page">Link</a>
{% endblock %}
```

then the `path/to/page` is just appended to the path that I currently have in my browser, leading to 404 not found on any non-starting page.

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @VladimirFokow,
should be possible, just like in the theme templates you can use the `url` filter to generate a relative link to the current page:
https://github.com/squidfunk/mkdocs-material/blob/6b239db64e031efe0399d338d3ce0a748c92a75e/src/templates/base.html#L102

The `'assets/stylesheets/main.css'` is passed to the `url` filter and processed accordingly.
https://github.com/mkdocs/mkdocs/blob/e755aaed7ea47348a60495ab364d5483ab90a4a6/mkdocs/utils/templates.py#L38
https://github.com/mkdocs/mkdocs/blob/e755aaed7ea47348a60495ab364d5483ab90a4a6/mkdocs/utils/__init__.py#L205

**Reply by VladimirFokow** [reply]
Thanks! It worked:

`main.html`:

```html
{% extends "base.html" %}

{% block announce %}
  <a href="{{ 'path/to/page' | url }}">Link</a>
{% endblock %}
```

where the `path/to/page` is relative to the `docs` directory

**Reply by kamilkrzyskow** [reply]
You're correct, I also saw another example in the file:
https://github.com/squidfunk/mkdocs-material/blob/6b239db64e031efe0399d338d3ce0a748c92a75e/src/templates/base.html#L53-L56

And the `page.url` is always relative to the docs with a `/` at the end when `use_directory_urls` is enabled, so `blog/2023/10/12/a-second-post/` etc.

**Comment by VladimirFokow** [comment]
maybe this trick can be surfaced to the [docs](https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-header/#announcement-bar)?

**Reply by kamilkrzyskow** [reply]
🤔 Perhaps and an explanation that absolute links using `{{ config.site_url ~ 'relative/path/' }}` should also work. 
cc @alexvoss

**Reply by alexvoss** [reply]
thanks, @kamilkrzyskow, that might be a good thing to add to the example in the documentation. Would relative URLs not be more robust, though? My understanding is that that is what the url filter would produce? 

Before we document this, we need to get clarity on what is and what is not part of the "API surface". I will check with squidfunk.
